src/main/java/kr/hhplus/be/server/infra/bestseller/redis/BestSellerRedisRepository.java

```diff
@@ -0,0 +1,93 @@
+package kr.hhplus.be.server.infra.bestseller.redis;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.stereotype.Repository;
+import org.springframework.util.CollectionUtils;
+
+import kr.hhplus.be.server.domain.bestseller.dto.BestSellerItem;
+import lombok.RequiredArgsConstructor;
+
+@Repository
+@RequiredArgsConstructor
+public class BestSellerRedisRepository {
+
+	private final RedisTemplate<String, Object> redisTemplate;
+
+	private static final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyyMMdd");
+	private static final String REDIS_KEY_PREFIX_RANKING = "bestseller:realtime:";
+	private static final String REDIS_KEY_PREFIX_PRODUCT_NAME = "bestseller:productname:";
+	private static final int MINIMUM_BESTSELLER_COUNT = 10; // 실시간 랭킹에서 보여줄 최소 개수
+
+	// 데이터를 유지할 일수
+	private static final int DATA_RETENTION_DAYS = 2;
+
+	public void incrementScore(BestSellerItem item) {
+		String productId = item.productId().toString();
+		String rankingKey = getRealTimeRankingKey();
+		String productNameKey = getRealTimeRankingProductNameKey();
+
+		redisTemplate.opsForZSet().incrementScore(rankingKey, productId, item.quantity());
+		redisTemplate.opsForHash().put(productNameKey, productId, item.productName());
+
+		// 2일 뒤 삭제한다. 다음 날의 실시간 랭킹 10개가 부족할 수 있으므로.
+		redisTemplate.expire(rankingKey, Duration.ofDays(DATA_RETENTION_DAYS));
+		redisTemplate.expire(productNameKey, Duration.ofDays(DATA_RETENTION_DAYS));
+	}
+
+	public List<String> getTodayTopProductNames(int limit) {
+		//  오늘 날짜 키에서 상위 상품들 조회
+		String todayKey = getRealTimeRankingKey();
+		String todayProductNameKey = getRealTimeRankingProductNameKey();
+		Set<Object> todayProductIds = redisTemplate.opsForZSet().reverseRange(todayKey, 0, limit - 1);
+
+		// 오늘 데이터가 10개 이상이면 오늘 데이터 사용
+		if (!CollectionUtils.isEmpty(todayProductIds) && todayProductIds.size() >= MINIMUM_BESTSELLER_COUNT) {
+			List<Object> productNames = redisTemplate.opsForHash()
+				.multiGet(todayProductNameKey, todayProductIds);
+
+			return productNames.stream()
+				.filter(Objects::nonNull)
+				.map(Object::toString)
+				.limit(limit)
+				.collect(Collectors.toList());
+		}
```

Comment:
```suggestion
		if (canUseTodayProducts(todayProductIds)) {
			List<Object> productNames = redisTemplate.opsForHash()
				.multiGet(todayProductNameKey, todayProductIds);

			return productNames.stream()
				.filter(Objects::nonNull)
				.map(Object::toString)
				.limit(limit)
				.collect(Collectors.toList());
		}
```

조건문을 위와 같이 메서드를 추출해보면 더 좋을 것 같네요?